source/module_io/numerical_basis_jyjy.cpp

```diff
@@ -0,0 +1,96 @@
+#include "module_io/numerical_basis_jyjy.h"
+#include "module_basis/module_nao/two_center_integrator.h"
+
+namespace NumericalBasis
+{
+#ifdef __LCAO
+using index_t = std::tuple<int, int, int, int>;
+
+std::vector<index_t> indexgen(const std::vector<int>& natom,
+                              const std::vector<int>& lmax)
+{
+#ifdef __DEBUG
+    assert(natom.size() == lmax.size());
+#endif
+    std::vector<index_t> index;
+    for (size_t itype = 0; itype < natom.size(); ++itype)
+    {
+        for (int iatom = 0; iatom < natom[itype]; ++iatom)
+        {
+            for (int l = 0; l <= lmax[itype]; ++l)
+            {
+                for (int M = 0; M < 2*l+1; ++M)
+                {
+                    // convert the "abacus M" to the conventional m
+                    int m = (M % 2 == 0) ? -M/2 : (M+1)/2;
```

Comment:
add a const for m?

Reply:
A const qualifiers has been added in the latest commit.